exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/flatten/FlattenRecordBatch.java

```diff
@@ -453,22 +453,15 @@ protected boolean setupNewSchema() throws SchemaChangeException {
         cg.addExpr(expr);
       } else {
         // need to do evaluation.
-        final MaterializedField outputField;
+        ValueVector incomingVector = null;
         if (expr instanceof ValueVectorReadExpression) {
           final TypedFieldId id = ValueVectorReadExpression.class.cast(expr).getFieldId();
-          @SuppressWarnings("resource")
-          final ValueVector incomingVector = incoming.getValueAccessorById(id.getIntermediateClass(), id.getFieldIds()).getValueVector();
-          // outputField is taken from the incoming schema to avoid the loss of nested fields
-          // when the first batch will be empty.
-          if (incomingVector != null) {
-            outputField = incomingVector.getField().clone();
-          } else {
-            outputField = MaterializedField.create(outputName, expr.getMajorType());
-          }
-        } else {
-          outputField = MaterializedField.create(outputName, expr.getMajorType());
+          incomingVector = incoming.getValueAccessorById(id.getIntermediateClass(), id.getFieldIds()).getValueVector();
```

Comment:
Are we sure that the dropped code is not needed? This looks like one of those fixes added after a session with the school of hard knocks. That is, added to fix some specific bug.

Can we track down the JIRA ticket to see what the problem is that the code solved and run a test to ensure that removing the code doesn't cause a regression?

Reply:
There is no dropped code, the code is re-organized to avoid nested if/else.